src/cloudWatchLogs/commands/searchLogGroup.ts

```diff
@@ -162,5 +181,163 @@ export class SearchLogGroupWizard extends Wizard<SearchLogGroupWizardResponse> {
+            label: 'Last 1 Hour',
+            data: 60,
+        })
+        options.push({
+            label: 'Last 12 Hours',
+            data: 60 * 12,
+        })
+        return options
+    }
+
+    public createMenuPrompter() {
+        const prompter = createQuickPick<integer | typeof customRange>(this.recentTimeOptions)
+
+        prompter.quickPick.items = [
+            ...prompter.quickPick.items,
+            {
+                label: 'Custom time range',
+                data: customRange,
+                detail: `YYYY/MM/DD-YYYY/MM/DD`,
+            },
+        ]
+
+        return prompter
+    }
+
+    private switchState(newState: 'custom-range' | 'recent-range') {
+        this.currentState = newState
+    }
+
+    public createDateBox(): InputBoxPrompter {
+        return createInputBox({
+            title: 'Enter custom date range',
+            placeholder: 'YYYY/MM/DD-YYYY/MM/DD',
+            validateInput: input => this.validateDate(input),
+        })
+    }
+
+    protected async promptUser(): Promise<PromptResult<TimeFilterResponse>> {
+        while (true) {
+            switch (this.currentState) {
+                case 'recent-range': {
+                    this.steps && this.defaultPrompter.setSteps(this.steps[0], this.steps[1])
+
+                    const resp = await this.defaultPrompter.prompt()
+                    if (resp === customRange) {
+                        this.switchState('custom-range')
+                    } else if (isValidResponse(resp)) {
+                        const [endTime, startTime] = [new Date(), new Date()]
+                        startTime.setHours(endTime.getHours() - resp)
+
+                        return { start: startTime.valueOf(), end: endTime.valueOf() }
+                    } else {
+                        return resp
+                    }
+
+                    break
+                }
+                case 'custom-range': {
+                    const resp = await this.customPrompter.prompt()
+                    if (isValidResponse(resp)) {
+                        const [startTime, endTime] = this.parseDate(resp)
+
+                        return { start: startTime.valueOf(), end: endTime.valueOf() }
+                    }
+
+                    this.defaultPrompter = this.createMenuPrompter() //reload the defaultPrompter
+                    this.switchState('recent-range')
+
+                    break
+                }
+            }
+        }
+    }
+
+    public validateDate(input: string) {
+        const parts = input.split('-')
+        const today = new Date()
+
+        if (parts.length !== 2) {
+            return 'String must include two dates seperated by `-`'
+        }
+        const [startTime, endTime] = parts
+
+        if (!Date.parse(startTime)) {
+            return 'starting time format is invalid, use YYYY/MM/DD'
+        }
+        if (!Date.parse(endTime)) {
+            return 'ending time format is valid, use YYYY/MM/DD'
```

Comment:
all of this logic could live in a shared common module, if we don't already have one.